### PR TITLE
upgrade dropwizard metrics

### PR DESCRIPTION
**Pull request by shamith09**

### CODE REVIEW

**Comment by wi11dey** [comment]
relevant
* https://github.com/dropwizard/metrics/issues/1753
* https://github.com/dropwizard/metrics/issues/1581

**Comment by wi11dey** [comment]
looks like this is the last remaining failure
```
    [junit] ERROR 22:29:41 Exception in response
    [junit] java.io.IOException: Connection reset by peer
    [junit] 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_171]
    [junit] 	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1140) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
    [junit] ERROR 22:29:41 Exception in response
    [junit] java.io.IOException: Connection reset by peer
    [junit] 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_171]
    [junit] 	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1140) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
    [junit] WARN  22:29:41 An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
    [junit] java.io.IOException: Connection reset by peer
    [junit] 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_171]
    [junit] 	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1140) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
    [junit] WARN  22:29:41 An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
    [junit] java.io.IOException: Connection reset by peer
    [junit] 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_171]
    [junit] 	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1140) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
    [junit] WARN  22:29:41 Cannot find Cassandra version for host localhost/127.0.0.1:38825 to parse the schema, using 1.2.0 based on protocol version in use. If parsing the schema fails, this could be the cause
    [junit] WARN  22:29:41 Cannot find Cassandra version for host localhost/127.0.0.1:38825 to parse the schema, using 1.2.0 based on protocol version in use. If parsing the schema fails, this could be the cause
    [junit] WARN  22:29:41 Cannot find Cassandra version for host localhost/127.0.0.1:38825 to parse the schema, using 2.0.0 based on protocol version in use. If parsing the schema fails, this could be the cause
    [junit] WARN  22:29:41 Cannot find Cassandra version for host localhost/127.0.0.1:38825 to parse the schema, using 2.0.0 based on protocol version in use. If parsing the schema fails, this could be the cause
    [junit] WARN  22:29:41 Cannot find Cassandra version for host localhost/127.0.0.1:38825 to parse the schema, using 2.2.0 based on protocol version in use. If parsing the schema fails, this could be the cause
    [junit] WARN  22:29:41 Cannot find Cassandra version for host localhost/127.0.0.1:38825 to parse the schema, using 2.2.0 based on protocol version in use. If parsing the schema fails, this could be the cause
    [junit] WARN  22:29:41 Cannot find Cassandra version for host localhost/127.0.0.1:38825 to parse the schema, using 2.1.0 based on protocol version in use. If parsing the schema fails, this could be the cause
    [junit] WARN  22:29:41 Cannot find Cassandra version for host localhost/127.0.0.1:38825 to parse the schema, using 2.1.0 based on protocol version in use. If parsing the schema fails, this could be the cause
    [junit] ERROR 22:29:43 Exception in response
    [junit] java.io.IOException: Connection reset by peer
    [junit] 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_171]
    [junit] 	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1140) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
    [junit] ERROR 22:29:43 Exception in response
    [junit] java.io.IOException: Connection reset by peer
    [junit] 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_171]
    [junit] 	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1140) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
    [junit] WARN  22:29:43 An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
    [junit] java.io.IOException: Connection reset by peer
    [junit] 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_171]
    [junit] 	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1140) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
    [junit] WARN  22:29:43 An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
    [junit] java.io.IOException: Connection reset by peer
    [junit] 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_171]
    [junit] 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_171]
    [junit] 	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1140) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.37.Final.jar:4.1.37.Final]
    [junit] 	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
    [junit] ------------- ---------------- ---------------
    [junit] Testcase: testIdleDisconnect(org.apache.cassandra.transport.IdleDisconnectTest):	FAILED
    [junit] 
    [junit] junit.framework.AssertionFailedError: 
    [junit] 	at org.apache.cassandra.transport.IdleDisconnectTest.testIdleDisconnect(IdleDisconnectTest.java:57)
    [junit] 
    [junit] 
    [junit] Test org.apache.cassandra.transport.IdleDisconnectTest FAILED
```

File: src/java/org/apache/cassandra/hadoop/cql3/CqlConfigHelper.java [hunk 1]
```diff
@@ -337,7 +340,12 @@ public static Cluster getCluster(String[] hosts, Configuration conf, int port)
                 .withQueryOptions(queryOptions)
                 .withPoolingOptions(poolingOptions);
 
-        return builder.build();
+        Cluster cluster = builder.build();
+
+        JmxReporter.forRegistry(cluster.getMetrics().getRegistry())
+                .inDomain(cluster.getClusterName() + "-metrics")
+                .build().start();
+        return cluster;
```
Comment (wi11dey) [inline src/java/org/apache/cassandra/hadoop/cql3/CqlConfigHelper.java]: flagging for later: potential leak of the jmx reporter here; we could make a subclass of Cluster that is a decorator and closes the reporter when the cluster is closed as well

File: test/unit/org/apache/cassandra/cql3/DeleteTest.java [hunk 2]
```diff
@@ -31,8 +31,7 @@
 import org.junit.BeforeClass;
 import org.junit.Test;
 
-public class DeleteTest extends SchemaLoader
-{
+public class DeleteTest extends SchemaLoader {
```
Comment (wi11dey) [inline test/unit/org/apache/cassandra/cql3/DeleteTest.java]: nit: before merging we should undo the whitespace changes like these

File: tools/stress/src/org/apache/cassandra/stress/util/JavaDriverClient.java [hunk 3]
```diff
@@ -22,6 +22,7 @@
 import java.util.concurrent.ConcurrentMap;
 import javax.net.ssl.SSLContext;
 
+import com.codahale.metrics.jmx.JmxReporter;
```
Comment (wi11dey) [inline tools/stress/src/org/apache/cassandra/stress/util/JavaDriverClient.java]: unnecessary import ?

File: src/java/org/apache/cassandra/hadoop/cql3/CqlConfigHelper.java (hunk 1 above)
Comment (shamith09) [reply src/java/org/apache/cassandra/hadoop/cql3/CqlConfigHelper.java]: Added Dropwizard4Cluster decorator class that closes reporter along with the cluster

File: src/java/com/palantir/cassandra/objects/Dropwizard4Cluster.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder extends Cluster.Builder {
+        @Override
+        public Cluster build() {
+            return new Dropwizard4Cluster(this);
```
Comment (wi11dey) [inline src/java/com/palantir/cassandra/objects/Dropwizard4Cluster.java]: if we do
```suggestion
            return new Dropwizard4Cluster(this.withoutJMXReporting());
```

then we can just replace all usages of Cluster with Dropwizard4Cluster and no other code changes

File: src/java/com/palantir/cassandra/objects/Dropwizard4Cluster.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package com.palantir.cassandra.objects;
+
+import com.codahale.metrics.jmx.JmxReporter;
+import com.datastax.driver.core.Cluster;
+
+public class Dropwizard4Cluster extends Cluster {
```
Comment (wi11dey) [inline src/java/com/palantir/cassandra/objects/Dropwizard4Cluster.java:11]: nit: would add a comment linking to https://docs.datastax.com/en/developer/java-driver/3.6/manual/metrics/index.html#metrics-4-compatibility for posterity

**Comment by wi11dey** [comment]
before merging, lets bump to latest (4.2.25) and have the only code changes be replacing all usages of Cluster.builder() with Dropwizard4Cluster.builder() (hoping that'll also get IdleDisconnectTest passing)

File: src/java/com/palantir/cassandra/objects/Dropwizard4Cluster.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    private JmxReporter reporter;
+
+    private Dropwizard4Cluster(Cluster.Builder builder) {
+        super(builder);
+        try {
+            this.reporter = JmxReporter.forRegistry(this.getMetrics().getRegistry())
+                    .inDomain(this.getClusterName() + "-metrics")
+                    .build();
+        } catch (NullPointerException e) {
+            // This is a workaround for yet unknown issue in the integration tests
```
Comment (wi11dey) [inline src/java/com/palantir/cassandra/objects/Dropwizard4Cluster.java]: we should run this down before merging

**Review by wi11dey — APPROVED** [review]
approving pending test

**Review by rhuffy — APPROVED** [review]

**Pull request merged**